hil/api.py

```diff
@@ -158,11 +158,12 @@ def project_detach_node(project, node):
         if (cfg.has_option('maintenance', 'shutdown')):
             node.obm.power_off()
         maintenance_proj.nodes.append(node)
-        response = requests.post(url, data=payload)
-        # Give warning if response > 400
-        if (response > 400):
-            logger.warn('POST to maintenance service' \
-                    ' failed with response: %s' % response)
+        response = requests.post(url,
+                                 headers={'Content-Type': 'application/json'},
+                                 data=payload)
+        if (response.status_code > 400):
```

Comment:
should be **>=**

or maybe just `>= 300` because that also is a failure, isn't it?

Reply:
Fixed.

Reply:
3xx aren't failures, they're things like redirects (which we should follow) "not modified" responses (which aren't applicable here) etc. I think requests is configured to follow redirects by default, but be should be defensive. Probably the thing to do is to check `200 <= response_code < 300`, and assume the library already follows redirects.